### DISCUSSION TITLE
What are the main differences between [Firenvim] and [GhostText with Neovim] ?

### DISCUSSION THREAD

**Discussion by adamency** [glacambre/firenvim#1566] [General]
Basically the title, it's great that you've linked other related projects, but now this begs me the question of what exactly firenvim does differently from GhostText that warrants a new independent project for this specific editor (neovim) instead of using the generic project ?

**Comment by alerque** [comment]
As a former GhostText user...

Totally and completely different beasts. Both of them save TextArea content to a system file and allow an external editor to edit it, but the similarities end there. GhostText launches an external process that results in a new window and an editor completely separate from the Firefox (or other browser) window.

On the other hand Firenvim launches an external editor process, but the Neovim instance it launches is a headless editor, and it renders a UI directly in the browser. This puts the editor interface in the browser tab in the same location on the page as the original TextArea element. This makes it much easier to not loose track of context when editing. With an external window you might not even have the right page open to match. With an embeded editor you have the original page and layout as designed and can even edit multiple inputs on the same page without wondering what is connected to what.

**Comment by adamency** [comment]
Ah yes, I seem to have not watched GhostText's readme screencast properly, I thought the editing was in the browser text area too, my mistake sorry. And thanks a lot for your prompt and detailed answer :)
